Refactor

adkuz opened the pull request.

Hamsterrhino committed:
Refactor

Hamsterrhino committed:
Refactor 2

Hamsterrhino committed:
Delete folder

Hamsterrhino committed:
Approven't

Hamsterrhino committed:
Changed system to full work in console, instead of line arguements.

Changed system to fully work in console, instead of line arguements.
Added command to show password in hidden way (shows *** + random amount of *)
Tried to add command to copy password to clipboard (Doesn't work right now)

Hamsterrhino committed:
A lot of polishing

Polished a lot of places to be more readable.
Tried to change copying to Clipboard, didn't work.

Hamsterrhino committed:
Fixed copying to Clipboard

Now copying to Clipboard works.
Note: if [Place] and [Login] have the same name, it will give the first entry in DB

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+
+#pragma warning(disable : 4996)
+
+unsigned char *toEncrypt(std::string word,
+                         unsigned char *key) // Encryption of strings
+{
+  unsigned char *encryptedword =
+      new unsigned char[strlen(word.c_str())]; // char* for encrypted password
+
+  unsigned char *tempword =
+      new unsigned char[strlen(word.c_str()) + 1]; // Copy of original password
+
+  strcpy((char *)tempword, word.c_str()); // Copying original password
+
+  // Encryption process
+  unsigned int outlen = 0;
+
+  AES encryp;
+  encryptedword = encryp.EncryptECB(tempword, strlen((char *)tempword) + 1, key,
+                                    outlen); // Encrypting password
+  return encryptedword;
+}
```
Не очень понимаю логику.

`std::string` - это c++
Однако строку копируешь через `strcpy` (си-шная штука). 
Почему так? Почему не решать более задачу на уровень выше, если уже пишешь на с++?

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+                                    outlen); // Encrypting password
+  return encryptedword;
+}
+unsigned char *toEncrypt(std::vector<char> word,
+                         unsigned char *key) // Encryption of vector<char>
+{
+  unsigned char *encryptedword =
+      new unsigned char[word.size()]; // char* for encrypted password
+
+  unsigned char *tempword =
+      new unsigned char[word.size() + 1]; // Copy of original password
+  for (size_t i = 0; i < word.size(); i++) {
+    tempword[i] = word[i];
+  }
+
+  // Encryption process
+  unsigned int outlen = 0;
+
+  AES encryp;
+  encryptedword = encryp.EncryptECB(tempword, strlen((char *)tempword) + 1, key,
+                                    outlen); // Encrypting password
+  return encryptedword;
+}
```
Зачем эта функция?

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+                                    outlen); // Encrypting password
+  return encryptedword;
+}
+unsigned char *
```
Не хватает проблема между функциями, а то очень неудобно читать.

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+  return decryptedword;
+}
+
+void save(std::vector<char *> data, sqlite3 *db, sqlite3_stmt *stmt,
+          char *err) // Code related to saving (name for a place, name for
+                     // login, password) into a database
```
Здесь функция возвращает void, имея один output-параметр `char* err` - зачем так?
Пусть функция возвращает указатель на строку ошибки, если ошибка есть, и `nullptr`, если функция выполнилась успешно.

Hamsterrhino replied:
` char* err ` Используется и пишет строку ошибки, но я понял что именно вы хотели.

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+
+  memcpy(key, keystring.c_str(), strlen(keystring.c_str()) + 1);
+  // Creating a table if it doesn't exist already
+  const char *sql =
+      _strdup("CREATE TABLE IF NOT EXISTS Passwords(SOURCE TEXT NOT "
+              "NULL, LOGIN TEXT,PASSWORD TEXT, DATE TEXT);");
+
```
Почему дата не timestamp?

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+      _strdup("CREATE TABLE IF NOT EXISTS Passwords(SOURCE TEXT NOT "
+              "NULL, LOGIN TEXT,PASSWORD TEXT, DATE TEXT);");
+
+  if (sqlite3_exec(db, sql, 0, 0, &err)) { // In case of error, we exit
+    std::cout << stderr << "Error SQL: " << err << std::endl;
```
Обработку ошибки следует сделать на уровень выше, то есть там, где вызывается функция `save`.

adkuz commented on Password Manager.cpp:
```diff
@@ -0,0 +1,436 @@
+         strlen(keystring.c_str()) + 1); // Copying key
+
+  unsigned char *source = (unsigned char *)data[0];
+  std::string sqstring = "SELECT * from Passwords WHERE SOURCE = @0;";
```
Вместо звездочки лучше явно указывать поля.

Hamsterrhino committed:
Loss of memory fixed

I tried to changed pointers to vectors, but it was harder than expected, so i manually deleted all pointers.

Hamsterrhino committed:
Fixed Alex's requests.

Hamsterrhino committed:
Deleteing useless stuff
